src/effects/deeplearning/ModelCard.h

```diff
@@ -0,0 +1,199 @@
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+   License: GPL v2.  See License.txt.
+
+   ModelCard.h
+   Hugo Flores Garcia
+
+******************************************************************/
+/**
+
+\class ModelCard
+\brief model metadata for deep learning models
+
+*/
+/*******************************************************************/
+
+#pragma once
+
+#include <torch/script.h>
+#include <torch/torch.h>
+
+#include <rapidjson/document.h>
+#include <rapidjson/schema.h>
+#include "rapidjson/prettywriter.h" 
+#include <rapidjson/writer.h>
+
+#include "MemoryX.h"
+#include "AudacityException.h"
+
+using DocHolder = std::shared_ptr<rapidjson::Document>;
+
+// this exception should be caught internally, but we 
+// derive from MessageBoxException just in case it needs to 
+// get handled by Audacity
+class InvalidModelCardDocument : public MessageBoxException
+{
+public:
+   InvalidModelCardDocument(const std::string& msg, 
+                           DocHolder doc)
+                           : m_msg(msg), m_doc(doc) ,
+                            MessageBoxException {
+                               ExceptionType::Internal,
+                               XO("Invalid Model Card Document")
+                            }
+   {}
+
+   // detailed internal error message
+   virtual const char* what() const throw () 
+   {
+      // TODO: also print the document
+      // TODO: check if document is nullptr
+      return m_msg.c_str();
+      // rapidjson::StringBuffer sb;
+      // rapidjson::Writer<rapidjson::StringBuffer> writer(sb);
+
+      // m_doc->Accept(writer);
+
+      // std::stringstream mMsg;
+      // mMsg << "Invalid ModelCard document: \n" << mMsg;
+      // mMsg << "document: " << std::string(sb.GetString()) << "\n";
+      // return mMsg.str();
+   }
+
+   // user facing message
+   virtual TranslatableString ErrorMessage() const
+      { return XO("An error occurred while validating model metadata.");}
+
+   const std::string m_msg;
+   DocHolder m_doc;
+};
+
+namespace parsers 
+{
+   DocHolder ParseString(const std::string &json);
+   DocHolder ParseFile(const std::string &path);
+}
+
+class DeepModelManager;
+
+class ModelCard
+{
+private:
+   friend class DeepModelManager;
+
+   ModelCard();
+   ModelCard(const ModelCard&);
+
+   // throws InvalidModelCardDocument if the given json is not valid. 
+   void DeserializeFromFile(const std::string& path, DocHolder schema);
+   void SerializeToFile(const std::string& path) const;
+
+   template < typename Writer >
+   void Serialize(Writer &writer) const;
+   void Deserialize(DocHolder doc, DocHolder schema);
+
+   void Validate(DocHolder doc, DocHolder schema);
+
+public:
+
+   bool is_local() { return m_is_local; }
```

Comment:
I would prefer sticking to CamelCase

Reply:
https://github.com/hugofloresgarcia/audacity/commit/f1371ccd74ea8a1b38a3f2c14fa30cc0200f4188